File: String/zigzagConversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1:
            return s
        
        depth = 1
        rows = 0
        v = [''] * numRows
        
        for i in s:
            v[rows] += i
            rows += depth
            
            if rows == numRows-1:
                depth = -1
            if rows == 0:
                depth = 1
            
        return  ''.join(v)

File: String/zigzagConversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        string result;
        if (numRows == 1)
            return s;
        
        vector<string>v(numRows,"");
        int depth = 1;
        int row = 0;
        for(auto c: s){
            v[row].push_back(c);
            row+= depth;
            if (row == numRows-1)
                depth = -1;
            if (row == 0)
                depth = 1;
        }
        for(auto val:v) result.append(val);
        return result;
    }
};
